### PR TITLE
[AIRFLOW-4239] Synchronize airflow execution timeout and aws batch job timeout

### PR DESCRIPTION
**Pull request by bovoleg**
Make sure you have checked _all_ steps below.

### Jira

- [x] https://issues.apache.org/jira/browse/AIRFLOW-4239

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by XD-DENG** [comment]
Hi @bovoleg , I haven't checked your code change yet. But may you read the PR template carefully and create JIRA then add details accordingly?

Your PR is a code change, which needs a JIRA to be created.

**Comment by bovoleg** [comment]
created Jira ticket AIRFLOW-4239
synchronize airflow task execution timeout and aws batch timeout
but not sure how to link with PR

**Review by ashb — CHANGES_REQUESTED** [review]
At the very least the parameter will need adding to the doc string, and it should mention the behaviour.

I'm not sure that setting the execution_timeout of the task to _exactly_ the same value is right either - that will stop the task a little before the timeout on Batch (due to other processing). Not sureif that is a problem though.

**Comment by bovoleg** [comment]
Hello,

   The “execution_timeout” argument is the parameter of the every airflow job. It is already well described in the documentation and in the BaseOperator class.
My suggested changes does not override its behavior, but just setup the AWS Batch execution timeout to the same value, which allows to synch task execution expectations between airflow and AWS Batch.
Without this changes Airflow may fail the task in 20 minutes when AWS batch may have timeout to 1 hours continues execution and success.

Oleg.


Від: Ash Berlin-Taylor <notifications@github.com>
Надіслано: 5 апреля 2019 г. 0:50
Кому: apache/airflow <airflow@noreply.github.com>
Копія: Oleg Burlachenko <oleg.burlachenko@zorallabs.com>; Mention <mention@noreply.github.com>
Тема: Re: [apache/airflow] [AIRFLOW-4239] Synchronize airflow execution timeout and aws batch job timeout (#5035)


@ashb requested changes on this pull request.

At the very least the parameter will need adding to the doc string, and it should mention the behaviour.

I'm not sure that setting the execution_timeout of the task to exactly the same value is right either - that will stop the task a little before the timeout on Batch (due to other processing). Not sureif that is a problem though.

File: airflow/contrib/operators/awsbatch_operator.py [hunk 1]
```diff
@@ -75,7 +76,7 @@ def __init__(self, job_name, job_definition, job_queue, overrides, max_retries=4
         self.job_queue = job_queue
         self.overrides = overrides
         self.max_retries = max_retries
-
+        self.execution_timeout = kwargs.get('execution_timeout', timedelta(seconds=3600))
```
Comment (Fokko) [inline airflow/contrib/operators/awsbatch_operator.py:79]: I would suggest not pulling this from the `kwargs` and make it an argument with a default value. WDYT?

**Comment by bovoleg** [comment]
The goal of this change is to setup AWS batch timeout to the same value as airflow batch timeout.

This value is already exists in the BatchOperator declaration
    :param execution_timeout: max time allowed for the execution of
        this task instance, if it goes beyond it will raise and fail.

So I think if add additional parameter we still may have a situation when AWS batch timeout and airflow timeout has different values, which I think we should avoid.

Oleg Burlachenko

Від: Fokko Driesprong <notifications@github.com>
Надіслано: 5 апреля 2019 г. 15:18
Кому: apache/airflow <airflow@noreply.github.com>
Копія: Oleg Burlachenko <oleg.burlachenko@zorallabs.com>; Mention <mention@noreply.github.com>
Тема: Re: [apache/airflow] [AIRFLOW-4239] Synchronize airflow execution timeout and aws batch job timeout (#5035)


@Fokko commented on this pull request.

________________________________

In airflow/contrib/operators/awsbatch_operator.py<https://github.com/apache/airflow/pull/5035#discussion_r272560358>:

> @@ -75,7 +76,7 @@ def __init__(self, job_name, job_definition, job_queue, overrides, max_retries=4

         self.job_queue = job_queue

         self.overrides = overrides

         self.max_retries = max_retries

-

+        self.execution_timeout = kwargs.get('execution_timeout', timedelta(seconds=3600))

I would suggest not pulling this from the kwargs and make it an argument with a default value. WDYT?

**Pull request closed without merging**